lib/Widgets/Settings.vala

```diff
@@ -88,6 +155,28 @@ namespace Granite {
             }
         }
 
+        private void setup_prefers_accent_color () {
+            try {
+                pantheon_act = GLib.Bus.get_proxy_sync (
+                    GLib.BusType.SYSTEM,
+                    "org.freedesktop.Accounts",
+                    user_path,
+                    GLib.DBusProxyFlags.GET_INVALIDATED_PROPERTIES
+                );
+
+                prefers_accent_color = (AccentColor) pantheon_act.prefers_accent_color;
+
+                ((GLib.DBusProxy) pantheon_act).g_properties_changed.connect ((changed, invalid) => {
+                    var accent_color = changed.lookup_value ("PrefersAccentColor", new VariantType ("i"));
+                    if (accent_color != null) {
+                        prefers_accent_color = (AccentColor) accent_color.get_int32 ();
+                    }
+                });
+            } catch (Error e) {
+                critical (e.message);
```

Comment:
It is very unlikely, but if `GLib.Bus.get_proxy_sync` fails, `prefers_accent_color` will return `null`.

I'm not sure which accent color is going to be the default one, but maybe we should fallback to it on error when `_prefers_accent_color == null`.

Reply:
I'm not quite sure. I simply copied and modified the code for `ColorScheme` 🤓️